File: LeetCode Questions + Tutorials/EngineeringDigest/Leetcode1189.ts
function maxNumberOfBalloons(text: string): number {
    let map = new Map<string, number>();
    for (let c of text) {
        if (c == 'b' || c == 'a' || c == 'l' || c == 'o' || c == 'n') {
            map.set(c, (map.get(c) || 0) + 1);
        }
    }
    let fCat: number = Math.min(map.get('b') || 0, Math.min(map.get('a') || 0, map.get('n') || 0));
    let sCat: number = Math.min(map.get('l') || 0, map.get('o') || 0);
    return Math.min(sCat / 2, fCat);
};

let text: string = "loonbalxballpoon";
console.log(maxNumberOfBalloons(text));

export { };

File: LeetCode Questions + Tutorials/EngineeringDigest/Leetcode1189.java
import java.util.*;

class Leetcode1189 {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.print("Enter the text: ");
        String txt = input.next();
        input.close();
        System.out.println("Total ballon: "+ maxNumberOfBalloons(txt));
    }
    private static int maxNumberOfBalloons(String text) {
        Map<Character, Integer> map = new HashMap();
        for(char c: text.toCharArray()){
            if(c == 'b' || c == 'a' || c == 'l' || c == 'o' || c == 'n') {
                map.put(c, map.getOrDefault(c, 0) + 1);
            }
        }
        int fCat = Math.min(map.getOrDefault('b', 0), Math.min(map.getOrDefault('a', 0), map.getOrDefault('n', 0)));
        System.out.println("fcat: "+ fCat);
        int sCat = Math.min(map.getOrDefault('l', 0), map.getOrDefault('o', 0));
        System.out.println("sCat: "+ sCat);
        return Math.min(sCat / 2, fCat);
    }
}